codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -314,12 +314,14 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
             byteDecoder = new PrefaceDecoder(ctx);
         }
         byteDecoder.channelActive(ctx);
+        super.channelActive(ctx);
     }
 
     @Override
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         if (byteDecoder != null) {
             byteDecoder.channelInactive(ctx);
+            super.channelInactive(ctx);
```

Comment:
@Scottmitch that would cause a `NullPointerException`  because byteDecoder is set to `null` (look at `ByteToMessageDecoder.channelInactive`, it calls `decode` again). I think it should be fine this way. To me, the bigger issue seems to be "what's all this `ByteDecoder` stuff", but let's not get into this argument :-).